pkg/user/password_rest.go

```diff
@@ -0,0 +1,38 @@
+package user
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/oklog/ulid"
+)
+
+type PasswordReset struct {
+	Token     string
+	CreatedAt time.Time
+	ExpiresAt time.Time
+	IsUsed    bool
```

Comment:
Instead of IsUsed, how about the approach of deleting the entire PasswordReset after the password reset is complete? This will make PasswordReset struct immutable